src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -209,9 +233,30 @@ class BlobCacheServiceImpl implements BlobCacheService {
      * @return The HTTP URI from the cached layer blob is going to be downloaded
      */
     protected String blobDownloadUri(RoutePath route) {
-        StringUtils.pathConcat(blobConfig.baseUrl, route.targetPath)
+        createPresignedGetUrl(blobConfig.storageBucket, route.targetPath)
```

Comment:
This is wrong. First `blobConfig.storageBucket` can be either `s3://some-bucket` or `s3://some-bucket/sub/path`. In both cases the [bucket()](https://github.com/seqeralabs/wave/blob/42b8f6bf438fa3534e2e0f9b3e1f804c57305f4a/src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy#L247-L247) method will fail (was this tested?) 

Second and more importantly, when `blobConfig.baseUrl` is provided it must be used to to compose the final URL. For example having: 
* targetPath: `some/container`
* storageBucket: `s3://foo/x` 
* baseUrl: `http://bar.com/y`

The signed URL will be something like `https://s3-region-foo.com/x/some/container?signature=1234&etc` 

The download URL should be `http://bar.com/y/some/container?signature=1234&etc`

Reply:
I have tested using
    storage:
      bucket: "s3://wave-dev-blob-cache"
      region: "eu-west-1"
      
      and got 
      `https://s3.EU-WEST-1.amazonaws.com/s3%3A%2F%2Fwave-dev-blob-cache/docker.io/v2/hrma017/dev/blobs/sha256%3Afba604e70bfeb8547d28bfb79f8bd825acca59804f2ec016045ecbda7ad92f46?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20240115T202800Z&X-Amz-SignedHeaders=host&X-Amz-Expires=1800&X-Amz-Credential=AKIAS3ISGBZV7QKVHBXT%2F20240115%2FEU-WEST-1%2Fs3%2Faws4_request&X-Amz-Signature=a5058b1b616f6ffb18a4e1e7ba66dcf88306c5755264f2b91b7b72a6c0a60b65`
      which i can see is wrong, i will correct it

Reply:
now the signed URL is 
`https://wave-dev-blob-cache.s3.EU-WEST-1.amazonaws.com/docker.io/v2/hrma017/dev/blobs/sha256%3A661ff4d9561e3fd050929ee5097067c34bafc523ee60f5294a37fd08056a73ca?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20240115T221942Z&X-Amz-SignedHeaders=host&X-Amz-Expires=1800&X-Amz-Credential=AKIAS3ISGBZV7QKVHBXT%2F20240115%2FEU-WEST-1%2Fs3%2Faws4_request&X-Amz-Signature=5e486693f78acfc43061ec0050fed8cf894d66ac1d353c6ef579ad9958de1d56`

Reply:
signed and download URL will be the same now

Reply:
Still not enough, if `baseUrl` is specified it should be used to compose the download URL \*and* the S3 signature (ie. all URL parameters after the `?` should be appended to this url). See the top of this thread

Reply:
done